cpp/terrain/terrain_outline.cpp

```diff
@@ -25,12 +28,10 @@ Texture *square_outline(coord::tile_delta foundation_size) {
 		}
 	}
 
-	Texture *ret = new Texture(width, height, image_data);
-	delete[] image_data;
-	return ret;
+	return std::make_shared<Texture>(width, height, image_data.get());
```

Comment:
Instead of passing a raw pointer to the image data, just add a Texture constructor that expects a unique_ptr to the image data and move image_data into the constructor. This avoids copying the complete image data within the constructor. The Texture's internal image data should be managed as unique_ptr anyway.